modules/tensor_mechanics/src/kernels/ADStressDivergenceRZTensors.C

```diff
@@ -0,0 +1,100 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADStressDivergenceRZTensors.h"
+#include "Assembly.h"
+#include "RankTwoTensor.h"
+#include "ElasticityTensorTools.h"
+#include "libmesh/quadrature.h"
+
+registerADMooseObject("TensorMechanicsApp", ADStressDivergenceRZTensors);
+
+defineADValidParams(
+    ADStressDivergenceRZTensors,
+    ADStressDivergenceTensors,
+    params.addClassDescription(
+        "Calculate stress divergence for an axisymmetric problem in cylindrical coordinates.");
+    params.addRequiredParam<unsigned int>(
+        "component",
+        "An integer corresponding to the direction the variable this kernel acts in. (0 "
+        "for x, 1 for y, 2 for z; note in this kernel x refers to the radial "
```

Comment:
Should we error if `_component != 0 && _component != 1`? You do this type of check in the constructor of the RSpherical version. Same comments regarding the duplicate parameters as above.